### PR TITLE
Plug fix

### PR DESCRIPTION
**Pull request by Cherser-s**
As I stated #1508 socket and plug didn't update outputs after dupe pasted. Also datasocket didn't set memory after being connected to the plug after latest commit. Also fixed coding style (removed C style logic operators).

### CODE REVIEW

**Comment by Divran** [comment]
I tested duping the dataplug and socket (while attached) a few minutes ago and it already worked fine

**Comment by Cherser-s** [comment]
Nope 
Try that with datasocket: wire memory inputs and outputs to the plug and socket for debug and then attach. Or do the same while attached and unplug it. Output won't change. My commit fixes it.

**Comment by Divran** [comment]
I can't reproduce this issue with the datasocket but I can with the normal socket, but only when it's not set to array mode.

**Comment by Cherser-s** [comment]
No on normal socket it works fine, except for dupe pasting case.

**Comment by Divran** [comment]
I just went onto my local server and set up a holo emitter that just had a blinking light, controlled through the datasocket by an E2. It kept blinking even after dupe. I'm pretty sure this is because the TriggerInput function is called after paste, which re-sets all necessary values. However, your PR won't hurt, plus it has some syntax cleanup so why not I guess?

File: lua/entities/gmod_wire_plug.lua [hunk 1]
```diff
@@ -62,7 +62,7 @@ end
 function ENT:Setup( ArrayInput )
 	self.ArrayInput = ArrayInput or false
 
-	if (!self.Inputs or !self.Outputs or self.ArrayInput != old) then
+	if not (self.Inputs and self.Outputs) then
```
Comment (Divran) [inline lua/entities/gmod_wire_plug.lua]: Why did you erase this check? The check for ArrayInput is necessary or  else the entity won't be updated properly if you left click it with the tool.

Comment (Cherser-s) [reply lua/entities/gmod_wire_plug.lua]: IDK how but couldn't save the file and commit that.

**Comment by Cherser-s** [comment]
Don't mind third commit, I somehow couldn't save changes. (got rid of optimisation)

**Pull request merged**